src/main/java/com/revature/studyforce/notification/service/StacktraceSubscriptionService.java

```diff
@@ -0,0 +1,136 @@
+     */
+    public StacktraceSubscription createStacktraceSubscription(Integer stacktraceId, Integer userId){
+        return stacktraceSubscriptionRepository.save(
+          new StacktraceSubscription(
+                  Objects.requireNonNull(stacktraceRepository.findById(stacktraceId).orElse(null)),
+                  subscriptionService.getSubscriptionByUserId(userId)
+          )
+        );
+    }
+
+    /**
+     * Deletes a stacktrace subscription for a given stacktrace and user subscription {@link StacktraceSubscriptionRepository#delete(Object)}
+     * The subscription is built by grabbing the stacktrace from {@link StacktraceSubscriptionRepository#findById(Object)}
+     * And the user subscription from {@link SubscriptionService#getSubscriptionByUserId(Integer)}
+     * @param stacktraceId The stacktraceId of the flashcard to be unsubscribed from
+     * @param userId The userId of the user to be unsubscribed
+     * @return The deleted stacktrace subscription.
+     */
+    public StacktraceSubscription deleteStacktraceSubscription(Integer stacktraceId, Integer userId){
+        StacktraceSubscription subscription = new StacktraceSubscription(
+          Objects.requireNonNull(stacktraceRepository.findById(stacktraceId).orElse(null)),
+          subscriptionService.getSubscriptionByUserId(userId)
+        );
+        stacktraceSubscriptionRepository.delete(subscription);
+        return subscription;
+    }
+
+    /**
+     * Deletes a list of subscriptions {@link StacktraceSubscriptionRepository#deleteAll(Iterable)}
+     * @param subscriptions The list of subscriptions to be deleted.
+     * @return The list of deleted subscriptions.
+     */
+    public List<StacktraceSubscription> deleteAllStacktraceSubscriptions(List<StacktraceSubscription> subscriptions){
+        stacktraceSubscriptionRepository.deleteAll(subscriptions);
+        return subscriptions;
+    }
+
+    /**
+     * Generates a list of subscriptions for a given stacktrace to be deleted by /deleteAllStacktraceSubscriptions
+     * @param stacktraceId The stacktraceId of the stacktrace to delete all subscribers.
+     * @return The list of deleted subscriptions.
+     */
+    public List<StacktraceSubscription> deleteAllStacktraceSubscriptionByStacktraceId(Integer stacktraceId){
+        List<StacktraceSubscription> subscriptions = getAllSubscribersByStacktraceId(stacktraceId);
+        return deleteAllStacktraceSubscriptions(subscriptions);
+    }
+
+    /**
+     * Generates a list of subscriptions for a given user to be deleted by /deleteAllStacktraceSubscriptions
+     * @param userId The userId of the user to delete all subscriptions.
+     * @return The list of deleted subscriptions.
+     */
+    public List<StacktraceSubscription> deleteAllStacktraceSubscriptionsByUserId(Integer userId){
+        List<StacktraceSubscription> subscriptions = getAllSubscriptionsByUserId(userId);
+        return deleteAllStacktraceSubscriptions(subscriptions);
+    }
+}
```

Comment:
Refactor with pagination
Check linting

Reply:
This is an internal service method, not something to be hit by an endpoint and served in the frontend.